test/extended/dns/dns.go

```diff
@@ -381,4 +460,92 @@ var _ = Describe("[sig-network-edge] DNS", func() {
 		pod := createDNSPod(f.Namespace.Name, cmd)
 		validateDNSResults(f, pod, expect, times)
 	})
+
+	It("should answer A and AAAA queries for a dual-stack service", func() {
+		// Only run this test on dual-stack enabled clusters.
+		networkConfig, err := oc.AdminConfigClient().ConfigV1().Networks().Get(context.Background(), "cluster", metav1.GetOptions{})
+		if err != nil {
+			e2e.Failf("unable to get cluster network config: %v", err)
+		}
+		usingIPv4 := false
+		usingIPv6 := false
+		for _, clusterNetworkEntry := range networkConfig.Status.ClusterNetwork {
```

Comment:
Is this the only way to determine a dual-stack?   What if it is dual-stack but only has ipv4 addresses for now?

Reply:
Clusters have to be configured to use both a v4 and v6 network on creation if dual-stack is desired, I believe. I don't think you can change from dualstack to single stack or vice versa after install time.

This is how we check if we are using dual-stack in the ingress operator, by the way:

https://github.com/openshift/cluster-ingress-operator/blob/master/pkg/operator/controller/ingress/deployment.go#L569-L588

Did you have a different way to check this in mind?

Reply:
Maybe you could check the Load Balancers for `spec.IpFamilyPolicy`, as mentioned in https://kubernetes.io/docs/concepts/services-networking/dual-stack/#service-type-loadbalancer

Reply:
Oop, never mind - the latter is what you did here.  Checking just the `clusterNetwork.CIDR`.  Looks fine.

Reply:
I think `networkConfig.Status.ClusterNetwork` could actually have more than just 2 entries. 

See https://github.com/openshift/cluster-network-operator/blob/master/pkg/network/cluster_config.go#L19

Reply:
https://github.com/openshift/cluster-network-operator/blob/3e9bc17218aa6dc6c3516264a301a2e118ceedc5/pkg/network/cluster_config.go#L46 shows it to be an error to have more than two.

Reply:
That's for the cluster's service network CIDR, not the cluster network CIDR.